### PR TITLE
Remove uniswap pairs redux

### PR DESCRIPTION
**Pull request by benisgold**
Fixes APP-821

## What changed (plus any additional context for devs)
Removed uniswap pairs redux. Why does this need to be stored in state at all? `rainbowTokenList` has state of its own that receives periodic updates. Instead of storing + accessing this data via redux state, we now directly access it from `rainbowTokenList`.

## Screen recordings / screenshots

https://github.com/rainbow-me/rainbow/assets/15272675/0420b408-3645-4141-927b-ae77e4012aa6

## What to test
1. Make sure the above logic is sound
2. Check that tokens are present in swap currency select modal + discover search

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -203,10 +200,6 @@ class OldApp extends Component {
     PerformanceContextMap.set('initialRoute', initialRoute);
   };
 
-  async handleTokenListUpdate() {
-    store.dispatch(uniswapPairsInit());
```
Comment (skylarbarrera) [inline src/App.js:207]: lmao :pain:

File: src/components/exchange/ExchangeAssetList.tsx [hunk 2]
```diff
@@ -297,7 +291,8 @@ const ExchangeAssetList: ForwardRefRenderFunction<
             setLocalFavorite(prev => {
               const address = rowData.address;
               const newValue = !prev?.[address];
-              dispatch(uniswapUpdateFavorites(address, newValue));
+              // dispatch(uniswapUpdateFavorites(address, newValue));
```
Comment (skylarbarrera) [inline src/components/exchange/ExchangeAssetList.tsx:294]: lets just delete this yeah?

File: src/hooks/useInitializeWallet.ts [hunk 3]
```diff
@@ -26,6 +25,7 @@ import { PROFILES, useExperimentalFlag } from '@/config';
 import { runKeychainIntegrityChecks } from '@/handlers/walletReadyEvents';
 import { checkPendingTransactionsOnInitialize } from '@/redux/data';
 import logger from '@/utils/logger';
+import { queryClient } from '@/react-query';
```
Comment (skylarbarrera) [inline src/hooks/useInitializeWallet.ts:28]: not being used?

File: src/model/migrations.ts [hunk 4]
```diff
@@ -674,10 +683,43 @@ export default async function runMigrations() {
 
   migrations.push(v17);
 
+  /**
+   *************** Migration v18 ******************
+   Move favorites from local storage to react query persistent cache (AsyncStorage)
+   */
+  const v18 = async () => {
```
Comment (skylarbarrera) [inline src/model/migrations.ts:690]: very much a nit but would love this to be in the new migrations flow our king @estrattonbailey set up , allows us to defer this so it doesn't impact app start up

**Review by skylarbarrera — APPROVED** [review]
looks good 2 me, tested with the reactQuery updates and things are working great 👌

File: src/model/migrations.ts (hunk 4 above)
Comment (estrattonbailey) [reply src/model/migrations.ts:690]: Miss you guysssss ❤️

Comment (benisgold) [reply src/model/migrations.ts:690]: ERIC!

File: src/hooks/useInitializeWallet.ts (hunk 3 above)
Comment (jinchung) [reply src/hooks/useInitializeWallet.ts:28]: looks like this gets removed in the other PR into this one: https://github.com/rainbow-me/rainbow/pull/5104/files#diff-53625aac5f03ab24c0c5622c32cefd9b214b500bd8131b1fcecfb01271f186b0

**Review by jinchung — APPROVED** [review]
🌮

**Pull request merged**